### PR TITLE
Enhance AdvancedCalendar Component with Range Selection, Date Presets, and Localization

### PR DESCRIPTION
**Pull request by AbdulrahmanNahhas**
## Description

This PR refines and enhances the AdvancedCalendar component by incorporating several new features that improve user experience and functionality. The key changes include:

- **Refined UI**: Improved the UI with Radix UI components, making it visually appealing and easier to interact with.
- **Support for Date Range Selection**: Users can now select a single date or a range of dates, enhancing the component’s flexibility for various use cases.
- **Presets for Quick Date Selection**: Added a set of predefined date options such as “Today,” “This Week,” “Last Week,” etc., to allow for quick and intuitive date selections.
- **Dynamic Localization**: The calendar automatically adapts to the user’s locale, displaying dates and labels appropriately.
- **Customizable Year Range**: Introduced fromYear and toYear props, allowing control over the range of selectable years.

These enhancements aim to create a more robust and user-friendly calendar experience.

## How has this been tested?

The changes have been tested through manual and automated testing methods:

1. Manual Testing: Verified the functionality of date selection (both single and range) in various scenarios. Ensured that the presets work as intended and the calendar displays the correct date based on user locale.
2. Browser Compatibility: Checked the component in multiple browsers to ensure consistent appearance and behavior.
3. Responsiveness: Tested on different screen sizes to confirm that the layout adjusts appropriately.

### Issue: https://github.com/ayasofyazilim-clomerce/ayasofyazilim-ui/issues/349

| Before | After |
|--------|-------|
|image|Screenshot 2024-10-26 at 8 55 53 PM|

### Issue: https://github.com/ayasofyazilim-clomerce/ayasofyazilim-ui/issues/348
Screenshot 2024-10-26 at 8 58 56 PM

### Feature: Two Months at the same time
Screenshot 2024-10-26 at 9 00 19 PM

### Features: Dates Buttons/Dropdown/Dropdown&Buttons:
Screenshot 2024-10-26 at 9 01 18 PM
Screenshot 2024-10-26 at 9 01 26 PM
Screenshot 2024-10-26 at 9 01 35 PM

### Presents
| Before | After |
|--------|-------|
|Screenshot 2024-10-26 at 9 04 58 PM|Screenshot 2024-10-26 at 9 05 38 PM|

### Feature: Localization
| English | Arabic | Turkish |
|--------|-------|-------|
|Screenshot 2024-10-26 at 9 09 20 PM|Screenshot 2024-10-26 at 9 08 56 PM|Screenshot 2024-10-26 at 9 10 04 PM|

## Types of changes

- [x] Chore (non-breaking change which improve code)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] Added story.
- [ ] Checked relative components.
- [ ] Checked the component on production.
- [ ] Added related docs.

### CODE REVIEW

File: .vscode/settings.json
Comment (a0m0rajab) [inline .vscode/settings.json:1]: I am not sure if we need this file, if you think it's helpful for the project, can you open an issue and a different PR for this to discuss it with the team?

Comment (a0m0rajab) [reply .vscode/settings.json:1]: Personally, I do not see a harm in having it.

File: @/components/ui/calendar.tsx [hunk 1]
```diff
@@ -1,70 +1,153 @@
...
+              </Select>
+            );
+          }
+        }
+        return null;
+      },
+      // eslint-disable-next-line react/no-unstable-nested-components
+      IconLeft: () => <ChevronLeftIcon className="h-4 w-4" />,
+      // eslint-disable-next-line react/no-unstable-nested-components
+      IconRight: () => <ChevronRightIcon className="h-4 w-4" />,
```
Comment (a0m0rajab) [inline @/components/ui/calendar.tsx:146]: We tend to follow the eslint standards, can you solve the issues provided in it? Here is a link about the issue: 
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md

File: @/components/ui/calendar.tsx [hunk 2]
```diff
@@ -1,70 +1,153 @@
...
+                  {currentMonth.getFullYear()}
+                </SelectTrigger>
+                <SelectContent>
+                  {selectedItems.map((selectItem) => (
+                    <SelectItem value={selectItem.value}>
+                      {selectItem.label}
+                    </SelectItem>
+                  ))}
+                </SelectContent>
+              </Select>
```
Comment (a0m0rajab) [inline @/components/ui/calendar.tsx:137]: this code is a duplicate of the previous one can we merge them together and have one return state for this component, I feel something like the next psudo code  world work:


```js
let itemsList = []
if (month) {
itemsList = doSomethingToFillIt() 
}
if (year) {
itemsList = fillit()
doSomething
}

return <SelectComponent>
```

Comment (a0m0rajab) [reply @/components/ui/calendar.tsx:137]: As for the setFullYear and seMonth, we can have a condition inside the onSelect Change. 
This will help us to maintain one select style for the months and years, if in the future wanted to change it we will have on place to edit.

File: src/molecules/advanced-calendar/advanced-calendar.stories.tsx
Comment (a0m0rajab) [inline src/molecules/advanced-calendar/advanced-calendar.stories.tsx:1]: why we deleted this file and changed the name?

File: src/molecules/calendar/calendar.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    <Calendar
+      {...args}
+      initialFocus
+      // onSelect={(value) => {
+      //   if ('toISOString' in value) {
+      //     setFilteredValue(value?.toISOString() || '');
+      //   }
+      // }}
+      // selected={filteredValue ? new Date(filteredValue) : undefined}
+    />
```
Comment (a0m0rajab) [inline src/molecules/calendar/calendar.stories.tsx:55]: We had the onSelect previously to get the state of the calendar externally, with the new implementation how can we achieve that? 

we have used it as well in the table/details-filters example, to understand how we expect it to work, the idea here is to be able to use this component with other components and get the value of it externally.

File: src/molecules/calendar/index.tsx
Comment (a0m0rajab) [inline src/molecules/calendar/index.tsx:1]: Looking at this file, I am not sure what changes you have done to it, can you keep the file names and do the changes on them? or if you would like to rename the file do not create a new file and work on it, this makes it hard to follow the changes for humans and Git.

**Review by a0m0rajab — CHANGES_REQUESTED** [review]
Thanks for the PR, this looks promising I just left few comments on it, feel free to check them and let me know if you have any questions. 

In general, I would suggest you to divide this PR so we can have a separate conversation for each feature, at the same time, it would be nice to check the other components that used a component if we changed it's name. For the calendar we have used in table and with this implementations it got broken.

File: .vscode/settings.json
Comment (AbdulrahmanNahhas) [reply .vscode/settings.json:1]: It's auto-generated by vscode

File: src/molecules/advanced-calendar/advanced-calendar.stories.tsx
Comment (AbdulrahmanNahhas) [reply src/molecules/advanced-calendar/advanced-calendar.stories.tsx:1]: I only changed the names:

1. `/advanced-calendar/advanced-calendar.stories.tsx` -> `/calendar/calendar.stories.tsx`
2. `/advanced-calendar/index.tsx` -> `/calendar/index.tsx`

File: @/components/ui/calendar.tsx
Comment (AbdulrahmanNahhas) [inline @/components/ui/calendar.tsx:1]: Here I solved issue: #349

File: src/molecules/calendar/index.tsx
Comment (AbdulrahmanNahhas) [reply src/molecules/calendar/index.tsx:1]: I'm sorry, I didn't know that would happend
